### PR TITLE
Update dynamics documentation

### LINKED ISSUE #12087: Update dynamics documentation
## Rationale

Documentation for dynamics was added to the moose wiki page. Moving it over to the new documentation system.

## Description

Added documentation for InertialForce and DynamicStressDivergenceTensors kernels, NewmarkVelAux and NewmarkAccelAux auxkernels, DynamicTensorMechanics Action and also added a general dynamics page that explains the theory.

Also corrected the jacobian bug in DynamicStressDivergenceTensors reported in #9431.

## Impact

Should not affected any of the tests.

### PR DESCRIPTION
**Pull request by sveerara**
Moved dynamics documentation from mooseframework wiki page to the new documentation system. 

Fixed bug in DynamicStressDivergenceTensors jacobian.

closes #12087 and #9431.

### CODE REVIEW

File: modules/tensor_mechanics/doc/content/modules/tensor_mechanics/Dynamics.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+\zeta &= \frac{\xi_t}{2 \pi} \; \frac{3}{(\Delta f)^2} \; \left(f_1 + f_2 - 2 \frac{f_1 f_1}{\Delta f} \; ln \frac{f_2}{f_1}\right) \\
+\eta &= 2 \pi \xi_t \; \frac{f_1 f_2}{\Delta_f} \; \left[ln \frac{f_2}{f_1}\; \left(2 + 6 \frac{f_1 f_2}{(\Delta_f)^2}\right) - \frac{3(f_1 + f_2)}{\Delta_f}\right]
+\end{aligned}
+\end{equation}
+
+A similar method can be used to estimate $\eta$ and $\zeta$ for non-constant damping ratios.
+
+## Implementation and Usage
+
+In the moose framework, the mass and stiffness matrices are not explicitly calculated. Only the residuals are calculated. To get the residual, the equation of motion (with Rayleigh damping and HHT time integration) can be written as:
```
Comment (bwspenc) [inline modules/tensor_mechanics/doc/content/modules/tensor_mechanics/Dynamics.md]: moose -> MOOSE

File: modules/tensor_mechanics/doc/content/modules/tensor_mechanics/Dynamics.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
+# Dynamics
+
+Dynamic problems like response of a multi degree of freedom structure to external forcing and wave propagation in a medium can also be solved using the Tensor Mechanics module.
```
Comment (bwspenc) [inline modules/tensor_mechanics/doc/content/modules/tensor_mechanics/Dynamics.md]: 'like response' -> 'like the response'

File: modules/tensor_mechanics/doc/content/modules/tensor_mechanics/Dynamics.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+\begin{equation}
+\rho \ddot{u}(t+\Delta t) + \eta \rho [(1+\alpha)\dot{u}(t + \Delta t)-\alpha \dot{u}(t)] + \zeta  \nabla \cdot [(1+\alpha)\frac{d}{dt}\sigma(t+\Delta t)- \alpha \frac{d}{dt}\sigma(t)] + \nabla \cdot [(1+\alpha) \sigma(t+\Delta t) - \alpha \sigma(t)] = F_{ext}(t+ (1 + \alpha) \Delta t)
+\end{equation}
+
+Here, $\rho$ is the density of the material and $\sigma$ is the stress tensor. The weak form of the above equation is used to get the residuals.
+
+The first two terms to the left ($ \rho \ddot{u}(t+\Delta t) + \eta \rho ((1+\alpha) \dot{u}(t + \Delta t)-\alpha \dot{u}(t)) $) are calculated in the [InertialForce](/InertialForce.md) kernel. $\alpha$, $\beta$, $\gamma$ and $\eta$ need to be passed as input to the InertialForce kernel.
+
+The next two terms to the left involving $\sigma$ are calculated in DynamicStressDivergenceTensors.C.
```
Comment (bwspenc) [inline modules/tensor_mechanics/doc/content/modules/tensor_mechanics/Dynamics.md]: Replace this with a link: `[DynamicStressDivergenceTensors](/DynamicStressDivergenceTensors.md)`

**Review by bwspenc — CHANGES_REQUESTED** [review]
Looks good.  Just a few minor things to touch up.

**Comment by sveerara** [comment]
@bwspenc , thanks for reviewing the PR. I have addressed your comments. I have also added an additional warning stating that DynamicStressDivergenceTensors uses undisplaced mesh whereas InertialForce kernel and Pressure BC use the displaced mesh. 

Bighorn, mammoth and rattlesnake tests are failing. I don't think those tests are failing because of this PR.

**Review by bwspenc — APPROVED** [review]

**Pull request merged**